chapters/chapter-abstracts.md

```diff
@@ -0,0 +1,147 @@
+
+It is critical to test software to reassure people (including yourself)
+that the software is correct enough
+and to make tolerances on "enough" explicit.
+There are several layers to testing software:
+adding assertions to code so that it checks itself as it runs;
+writing unit tests to check individual pieces of code;
+writing integration tests to check that those pieces work together correctly;
+writing regression tests to check if things that used to work no longer do.
+Fortunately,
+there are many excellent test frameworks to aid us with these tasks.
+A test framework finds and runs tests written in a prescribed fashion
+and reports their results.
+In addition to test failures,
+these reports often include the tests' coverage,
+which is the fraction of lines of code that are checked.
+To automate the execution of our tests
+we can use Continuous Integration
+which runs tests every time we make changes to the code,
+so that we can immediately spot when something breaks.
+
+## 12. Handling errors
+
+People will misunderstand how to use our programs,
+and we should therefore plan from the start to detect and handle errors.
+Something that goes wrong while a program is running
+is sometimes referred to as an exception from normal behavior.
+Generally speaking,
+we distinguish between two types of errors/exceptions.
+Internal errors are mistakes in the program itself,
+such as calling a function with None instead of a list.
+External errors are usually caused by interactions
+between the program and the outside world:
+a user may mis-type a filename,
+the network might be down,
+and so on.
+
+## 13. Tracking Provenance
+
+Modern publishing involves much more than producing a PDF
+and making it available on a preprint server.
+It also entails providing the data underpinning the report
+and the code used to do the analysis.
+While some reports, datasets, software packages, and analysis scripts
+can't be published without violating personal or commercial confidentiality,
+every researcher's default should be to make all these as widely available as possible.
+Publishing it under an open license is the first step.
+Using DOIs facilitates identification of reports, datasets, and software releases.
+Similarly,
+using an ORCID ensures that your research profile it correctly linked with your work
+and not mixed up with others of the same name.
+Any published data should be FAIR:
+findable, accessible, interoperable, and reusable.
+While small datasets can be put under standard version control;
+large ones should be stored on data sharing sites,
+sometimes there are specific ones for you field or the technology used for data collection.
+To facilitate for others to use your software, it is important that you
+describe your software environment, analysis scripts, and data processing steps
+in reproducible and easily inspectable ways.
+
+## 14. Creating Packages with Python
+
+While the core of programming languages are useful on their own,
```

Comment:
```suggestion
While the libraries that come standard with a programming language are useful on their own,
```

Reply:
I would remove the word 'standard' from Damien's suggested fix.